packages/firebase_storage/android/src/main/java/io/flutter/plugins/firebase/storage/FirebaseStoragePlugin.java

```diff
@@ -109,6 +118,24 @@ public void onFailure(@NonNull Exception e) {
             });
   }
 
+  private void getBucket(MethodCall call, final Result result) {
+    String path = call.argument("path");
+    StorageReference ref = firebaseStorage.getReference().child(path);
+    result.success(ref.getBucket());
+  }
+
+  private void getName(MethodCall call, final Result result) {
+    String path = call.argument("path");
+    StorageReference ref = firebaseStorage.getReference().child(path);
+    result.success(ref.getName());
+  }
+
+  private void getPath(MethodCall call, final Result result) {
+    String path = call.argument("path");
+    StorageReference ref = firebaseStorage.getReference().child(path);
+    result.success(ref.getPath());
```

Comment:
Is `ref.getPath()` ever different from the `path` provided in the argument?

Reply:
yes it might be, as discussed.